### ISSUE TITLE
`NoNameShadowing` false positive on `_`

### ISSUE THREAD

**Issue by rschattauer** [detekt/detekt#8962]
## Observed Behavior
Let's take this example code:

```
something.onClick { _, used ->
    try {
        used.doSomething()
    } catch (_: SomeSpecificException) {
        used.doSomethingElse()
    }
}
```
Then the second `_` is falsely marked as name shadowing. NameShadowing shouldn't work on `_` names.

## Your Environment
* Version of detekt used: 2.0.0-alpha.1

**Comment by schalkms** [comment]
Agreed.
The rule `NoNameShadowing` should ignore `_` names. Docs should also be updated.

**Comment by rschattauer** [comment]
I raised a PR. Is this according to what you want? :)

**Comment by schalkms** [comment]
Closed by 57f0726
